lib/terraforming/resource/datadog_monitor.rb

```diff
@@ -0,0 +1,128 @@
+        ENV["DATADOG_APP_KEY"]
+      end
+
+      def self.tf(client: Dogapi::Client.new(self.api_key, self.app_key))
+        self.new(client).tf
+      end
+
+      def self.tfstate(client: Dogapi::Client.new(self.api_key, self.app_key))
+        self.new(client).tfstate
+      end
+
+      def initialize(client)
+        @client = client
+      end
+
+      def tf
+        apply_template(@client, "tf/datadog_monitor")
+      end
+
+      def tfstate
+        monitors.inject({}) do |resources, monitor|
+          resources["datadog_monitor.#{module_name_of(monitor)}"] = {
+            "type" => "datadog_monitor",
+            "depends_on" => [],
+            "primary" => {
+              "id" => monitor["id"].to_s,
+              "attributes" => monitor_attributes(monitor),
+              "meta" => {},
+              "tainted" => false
+            },
+            "deposed" => [],
+            "provider" => ""
+          }
+          resources
+        end
+      end
+
+      private
+
+      def monitors
+        response = check_response(@client.get_all_monitors)
+
+        # Fix missing thresholds
+        response.each do |monitor|
+          if monitor["options"]["thresholds"].nil?
+            critical = monitor["query"].split.last
+            monitor["options"]["thresholds"] = { "critical" => critical }
+          else
+            monitor["options"]["thresholds"].each { |k, v| monitor["options"]["thresholds"][k] = v.round }
+          end
+        end
+
+        response
+      end
+      
+      def check_response(response)
+        case response[0]
+        when "401"
+          raise "Access Unauthorized, Datadog responded with a 401"
+        when "403"
+          raise "Access Forbidden, Datadog responded with a 403"
+        when "500"
+          raise "Server Error, Datadog responded with a 500"
+        end
+
+        response[1]
+      end
+
+      def module_name_of(monitor)
+        normalize_module_name(monitor["name"])
+      end
+
+      def monitor_attributes(monitor)
+        attributes = {
+          "id" => monitor["id"],
+          "name" => monitor["name"],
+          "type" => monitor["type"],
+          "message" => monitor["message"],
+          "escalation_message" => monitor["escalation_message"],
+          "query" => monitor["query"],
+          "notify_no_data" => monitor["options"]["notify_no_data"],
+          "no_data_timeframe" => monitor["options"]["no_data_timeframe"],
+          "renotify_interval" => monitor["options"]["renotify_interval"],
+          "notify_audit" => monitor["options"]["notify_audit"],
+          "timeout_h" => monitor["options"]["timeout_h"],
+          "include_tags" => monitor["options"]["include_tags"],
+          "require_full_window" => monitor["options"]["name"],
+          "locked" => monitor["options"]["locked"]
+        }
+        %w(silenced thresholds).each do |argument|
+            option = monitor["options"][argument]
```

Comment:
__[rubocop]__
* Use 2 (not 4) spaces for indentation. [:ref](https://github.com/bbatsov/ruby-style-guide#spaces-indentation)

<!-- Comment by [SideCI](https://sideci.com) -->